owners/test/owners_bot.test.js

```diff
@@ -215,11 +223,21 @@ describe('owners bot', () => {
           '_test_hash_',
           checkRun
         );
-
         // Ensures the test fails if the assertion is never run.
         expect(true).toBe(true);
       });
     });
+
+    it('creates a notification comment', async () => {
+      expect.assertions(1);
+      getCheckRunIdsStub.returns({});
+      sandbox.stub(OwnersBot.prototype, 'createNotifications');
+      await ownersBot.runOwnersCheck(github, pr);
+
+      sandbox.assert.calledOnce(ownersBot.createNotifications);
```

Comment:
What if we used `expect(ownersBot.createNotifications).toHaveBeenCalled(1)`? Could we remove `expect.assertions(1)` and `expect(true).toBe(true)`?

For reference, I use this when testing async code in the pr-deploy tests.

Reply:
Throughout the project, both when Erwin owned it and after I took over, we've been using `sinon` sandboxes for mocking and stubbing. I'm not inclined to mix in Jest mock functions in this PR, and this is not the first instance of the `expect(true).toBe(true)` pattern. We could go through and rewrite all the tests with `jest` mocks and remove `sinon` (assuming Jest provides similar sandboxing features, which I'm not certain of), which would let us use 'expect...toHaveBeenCalled`, but the documentation on that generally seems to indicate it's for use with explicitly mocked callback functions whose invocations need to be validated.

Reply:
A rewrite sounds time consuming, but do you agree that these patterns can be improved?
1. Using 2 types of assertions (jest and sinon) in a test suite
2. Using `expect(true).toBe(true)` to determine that `sinon.assert.calledOnce(spy)` ran

Reply:
Sinon is designed to be used in another test suite, be it mocha or jest or something else, so I don't see an issue with that. I don't like the expects there, but sinon doesn't have a similar expect.assertions method. I'm not sure what the best solution would be.

Reply:
One solution would be to use jest mocks and jest assertions. What do you think?

Reply:
#447

Reply:
Tests added in this PR have been updated. The other tests in this project will be updated in a later PR

Reply:
#454

Reply:
Thanks for following up!

Reply:
Thanks for pointing it out, would never have even found that this feature exists